src/middlewared/middlewared/plugins/smart.py

```diff
@@ -387,10 +387,16 @@ async def manual_test(self, disks):
                         **disk
                     })
                 else:
-                    verrors.add(
-                        f'disks.{index}.identifier',
-                        f'{disk["identifier"]} is not valid. Please provide a valid disk identifier.'
-                    )
+                    if await self.middleware.call('disk.query', [('identifier', '=', disk['identifier'])]):
```

Comment:
We should probably avoid calling `disk.query` for each disk and rather call this once above and save the result and consume them or maybe otherwise use `force_sql_filters` in query which would optimize this call.
```
❯ midclt call disk.query '[["identifier", "=", "{uuid}0f723d8d-29a9-4296-b215-a8723aac88c2"]]' '{"extra": {"force_sql_filters": true}}'
```